### PR TITLE
Fix: Make entryFile optional if controllerPathGlobs given

### LINKED ISSUE #786: entryFile should be optional if controllerPathGlobs is given
## Sorting

- **I'm submitting a ...**

  - [x] bug report
  - [ ] feature request
  - [ ] support request

- I confirm that I
  - [x] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Expected Behavior

`"entryFile"` and `"controllerPathGlobs"` in `tsoa.json` should be mutually exclusive such that only one of them must be given.

## Current Behavior

Both `"entryFile"` and `"controllerPathGlobs"` are required. I currently provide an empty TS file as entry file and everything still works.

## Possible Solution

Allow `entryFile` to be empty if `controllerPathGlobs` is given.

## Steps to Reproduce

See above.

## Context (Environment)

Version of the library: 3.3.0-alpha.0

### PR DESCRIPTION
**Pull request by aerabi**
Fixes #786

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/lukeautry/tsoa/tree/master/docs/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lukeautry/tsoa/pulls) for the same update/change?
- [x] Have you written unit tests?
- [x] Have you written unit tests that cover the negative cases (i.e.: if bad data is submitted, does the library respond properly)?
- [x] This PR is associated with an existing issue?

**Closing issues**

Put `closes #XXXX` (where XXXX is the issue number) in your comment to auto-close the issue that your PR fixes.

### If this is a new feature submission:

- [ ] Has the issue had a maintainer respond to the issue and clarify that the feature is something that aligns with the [goals](https://github.com/lukeautry/tsoa#goal) and [philosophy](https://github.com/lukeautry/tsoa#philosophy) of the project?

**Potential Problems With The Approach**

**Test plan**

### CODE REVIEW

**Comment by aerabi** [comment]
So, I could not find any test for other parts of the `cli.ts`. Should I create some?

File: packages/cli/src/cli.ts [hunk 1]
```diff
@@ -148,8 +148,8 @@ export interface ExtendedRoutesConfig extends RoutesConfig {
 }
 
 const validateRoutesConfig = async (config: Config): Promise<ExtendedRoutesConfig> => {
-  if (!config.entryFile) {
-    throw new Error('Missing entryFile: Configuration must contain an entry point file.');
+  if (!config.entryFile && !config.controllerPathGlobs) {
+    throw new Error('Missing entryFile and controllerPathGlobs: Configuration must contain an entry point file or controller path globals.');
   }
   if (!(await fsExists(config.entryFile))) {
```
Comment (WoH) [inline packages/cli/src/cli.ts:154]: This will now fail.

File: packages/cli/src/cli.ts [hunk 2]
```diff
@@ -148,8 +148,8 @@ export interface ExtendedRoutesConfig extends RoutesConfig {
 }
 
 const validateRoutesConfig = async (config: Config): Promise<ExtendedRoutesConfig> => {
-  if (!config.entryFile) {
-    throw new Error('Missing entryFile: Configuration must contain an entry point file.');
+  if (!config.entryFile && !config.controllerPathGlobs) {
```
Comment (WoH) [inline packages/cli/src/cli.ts:151]: Need to apply similar changes here: https://github.com/lukeautry/tsoa/blob/99502d0979e439b0b8ccfc6a58eca655022c8109/packages/cli/src/cli.ts#L94

**Review by WoH — COMMENTED** [review]
I don't think we need to test this, but there are some other places, where we need to change things up.

**Comment by aerabi** [comment]
So, it should be fine now.

**Review by WoH — APPROVED** [review]

**Comment by WoH** [comment]
You're correct, do you want to send a PR?

**Pull request merged**